app/controllers/repositories_controller.rb

```diff
@@ -212,4 +221,64 @@ def check_copy_permissions
   def repository_params
     params.require(:repository).permit(:name)
   end
+
+  def generate_zip
+    # Fetch rows in the same order as in the currently viewed datatable
+    ordered_row_ids = params[:row_ids]
```

Comment:
Sorry, but I'm double-checking: could a malicious user inject any IDs into this array before submitting array to retrieve rows from other repositories in other teams he's not a part of? I think here you need to check that only rows from repositories that belong to current team/repository (this is retrieved from route anyway right?) shall be queried.